snapshotter/demux/metrics/proxy.go

```diff
@@ -100,5 +172,10 @@ func (mp *Proxy) metrics(w http.ResponseWriter, req *http.Request) {
 	defer res.Body.Close()
 	response := io.LimitReader(res.Body, MaxMetricsResponseSize)
 
-	io.Copy(w, response)
+	_, err = io.Copy(w, response)
+	if err != nil {
+		log.G(ctx).Errorf("error writing response body: %v\n", err)
+		w.WriteHeader(500)
+		return
+	}
```

Comment:
Will the newline look strange in logs?

```suggestion
    if _, err := io.Copy(w, response); err != nil {
        log.G(ctx).Errorf("error writing response body: %v", err)
        w.WriteHeader(500)
    }
```

Reply:
Gp, I can remove the newline 👍